### PR TITLE
Decoder.getEntry() may returns NULL

### PR DESCRIPTION
**Pull request by lifpan**
This happens when SPIR-V binary contains useless word which does not belong to
any instruction. Its high 16-bit is 0, and the decoder discards it.

### CODE REVIEW

**Comment by lifpan** [comment]
Here is one SPIR-V binary generated by glslangValidator.exe (KhronosGroup/glslang), and the original GLSL shader.
[ObjInput_TestIndirect.zip](https://github.com/KhronosGroup/SPIRV-LLVM/files/1400903/ObjInput_TestIndirect.zip)

llvm-spirv -r ObjInput_TestIndirect.spv

llvm-spirv does not accept "GLSL.std.450", so please manually skip it:
          %1 = OpExtInstImport "GLSL.std.450"

**Comment by bader** [comment]
@lifpan, is it possible to add the test to SPIR-V lit test suite?
Here is an example of the SPIR-V test in this test suite:
https://github.com/KhronosGroup/SPIRV-LLVM/blob/khronos/spirv-3.6.1/test/SPIRV/vector_times_scalar.spt

Note: you should put run commands at the end of SPIR-V text file.
Probably you don't need to run FileCheck - only llvm-spirv.

File: test/SPIRV/useless_word.spt
Comment (bader) [inline test/SPIRV/useless_word.spt]: Just out of curiosity. Which instruction causes the issue?
```
0 Capability
```
or 
```
0 Nop
```
Both?

**Review by bader — APPROVED** [review]
LGTM.

File: test/SPIRV/useless_word.spt
Comment (lifpan) [reply test/SPIRV/useless_word.spt]: Both. By the way, shall we name it "redundant_word.spt"?

Comment (bader) [reply test/SPIRV/useless_word.spt]: Sound good to me.
Feel free to update the request.
I'll let Alexey S and Sam to take a look before the merge.
Thanks!

File: lib/SPIRV/libSPIRV/SPIRVModule.cpp
Comment (AlexeySotkin) [inline lib/SPIRV/libSPIRV/SPIRVModule.cpp]: It is common to use nullptr instead of NULL here. Please update.
Other than that, LGTM. Thanks.

**Comment by yxsamliu** [comment]
LGTM. Thanks!

**Pull request merged**
